skyfield/elementslib.py

```diff
@@ -9,17 +9,28 @@
                    sinh, sqrt, arccos, arctanh, zeros_like, ones_like, divide,
                    where, pi, cross)
 
-def osculating_elements_of(position, reference_frame=None):
+def osculating_elements_of(position, reference_frame=None, gm_km3_s2=None):
     """Produce the osculating orbital elements for a position.
 
-    The ``position`` should be an :class:`~skyfield.positionlib.ICRF`
-    instance like that returned by the ``at()`` method of any Solar
-    System body, specifying a position, a velocity, and a time.  An
-    instance of :class:`~skyfield.elementslib.OsculatingElements` is
-    returned.
+    Parameters:
+    position : :class:`~skyfield.positionlib.ICRF` object
+         commonly returned by the ``at()`` method of any Solar System body
+    reference_frame: 3x3 numpy array, optional
+        commonly used values are found in skyfield.data.spice.inertial_frames
+    gm_km3_s2: float, optional
+        Gravitational parameter (G*M) in units of km^3/s^2 If not specified, this is calculated for you.
 
+    Returns:
+    :class:`~skyfield.elementslib.OsculatingElements`
     """
-    mu = GM_dict.get(position.center, 0) + GM_dict.get(position.target, 0)
+    if not gm_km3_s2:
+        gm_km3_s2 = GM_dict.get(position.center, 0)
```

Comment:
Put “0.0” here, which looks like a GM value, rather than “0”, which at first reading I thought was a default index into `GM_dict` (I thought you were pulling the barycenter mass, for a moment). Plus of course it saves Python from having to convert integer to float later. Also in your other `.get()`.